com.microsoft.mrtk.input/Visualizers/ControllerVisualizer/ControllerVisualizer.cs

```diff
@@ -0,0 +1,183 @@
+                // OR the input device is specifically a simulated controller that is in the MotionController Simulation Mode.
+                bool useFallbackVisuals;
+                bool isSimulatedController;
+                if (xrInputDevice is MRTKSimulatedController simulatedController)
+                {
+                    useFallbackVisuals = simulatedController.SimulationMode == ControllerSimulationMode.MotionController;
+                    isSimulatedController = true;
+                }
+                else
+                {
+                    useFallbackVisuals = !HandsUtils.GetSubsystem().TryGetJoint(TrackedHandJoint.Palm, handNode, out _);
+                    isSimulatedController = false;
+                }
+                InstantiateControllerVisuals(!isSimulatedController, useFallbackVisuals);
+            }
+        }
+
+        // Private reference to the gameobject which represents the visualized controller
+        // Needs to be explicitly set to null in cases where no controller visuals are ever loaded.
+        private GameObject ControllerGameObject = null;
+
+        /// <summary>
+        /// Tries to instantiate controller visuals for the specified hand node.
+        /// </summary>
+        /// <param name="usePlatformVisuals">Whether or not to try to load visuals from the platform provider</param>
+        /// <param name="useFallbackVisuals">Whether or not to use the fallback controller visuals</param>
+        private async void InstantiateControllerVisuals(bool usePlatformVisuals, bool useFallbackVisuals)
+        {
+            // Disable any pre-existing controller models before trying to render new ones.
+            platformLoadedGameObject?.SetActive(false);
+            platformLoadedGameObjectRoot?.SetActive(false);
+            fallbackGameObject?.SetActive(false);
+
+            // Try to load the controller model from the platform
+            if (usePlatformVisuals)
+            {
+                platformLoadedGameObject = await ControllerModelLoader.TryGenerateControllerModelFromPlatformSDK(handNode.ToHandedness());
+                if (platformLoadedGameObject != null)
+                {
+                    // Platform models are "rotated" 180 degrees due to the forward vector for a controller pointing towards the user.
+                    // We need to rotate these models in order to have them pointing in the correct direction on device.
+                    if (platformLoadedGameObjectRoot == null)
+                    {
+                        platformLoadedGameObjectRoot = new GameObject("Platform Model Root");
+                    }
+                    platformLoadedGameObject.transform.parent = platformLoadedGameObjectRoot.transform;
+                    platformLoadedGameObject.transform.SetPositionAndRotation(platformLoadedGameObjectRoot.transform.position, platformLoadedGameObjectRoot.transform.rotation * Quaternion.Euler(0, 180, 0));
```

Comment:
Why the rotation offset? That seems sort of weird (and might deserve a const somewhere)

Reply:
It's because the loaded Platform models are "rotated" 180 degrees due to the forward vector for a controller pointing towards the user. We need to rotate these models in order to have them pointing in the correct direction on device.

Reply:
Done